docs/source/1.0/spec/aws/aws-cloudformation.rst

```diff
@@ -0,0 +1,672 @@
+
+.. tabs::
+
+    .. code-tab:: smithy
+
+        namespace smithy.example
+
+        use aws.cloudformation#additionalIdentifier
+        use aws.cloudformation#resource
+
+        @resource
+        resource Foo {
+            identifiers: {
+                fooId: String,
+            },
+            get: GetFoo,
+        }
+
+        operation GetFoo {
+            input: GetFooRequest,
+        }
+
+        structure GetFooRequest {
+            @required
+            fooId: String,
+
+            @additionalIdentifier
+            fooAlias: String,
+        }
+
+
+.. _aws.cloudformation#propertyName-trait:
+
+-----------------------------------------
+``aws.cloudformation#propertyName`` trait
+-----------------------------------------
+
+Summary
+    The propertyName trait allows a CloudFormation resource property name to
+    differ from a structure member name used in the model.
+Trait selector
+    ``structure > member``
+
+    *Any structure member*
+Value type
+    ``string``
+
+Given the following structure definition that is derived in to a CloudFormation
+resource:
+
+.. tabs::
+
+    .. code-tab:: smithy
+
+        namespace smithy.example
+
+        use aws.cloudformation#propertyName
+
+        structure AdditionalFooProperties {
+            bar: String,
+
+            @propertyName("Tags")
+            tagList: TagList,
+        }
+
+the CloudFormation resource would have the following property names derived
+from it:
+
+::
+
+    "bar"
+    "Tags"
+
+
+.. _aws.cloudformation#excludeProperty-trait:
+
+--------------------------------------------
+``aws.cloudformation#excludeProperty`` trait
+--------------------------------------------
+
+Summary
+    Indicates that structure member should not be included in generated
+    CloudFormation resource definitions.
+Trait selector
+    ``structure > member``
+
+    *Any structure member*
+Value type
+    Annotation trait
+Conflicts with
+    :ref:`aws.cloudformation#additionalIdentifier-trait`,
+    :ref:`aws.cloudformation#createOnlyProperty-trait`,
+    :ref:`aws.cloudformation#mutableProperty-trait`,
+    :ref:`aws.cloudformation#readOnlyProperty-trait`,
+    :ref:`aws.cloudformation#writeOnlyProperty-trait`
+
+The following example defines a CloudFormation resource that has the derivable
+``responseCode`` property excluded:
+
+.. tabs::
+
+    .. code-tab:: smithy
+
+        namespace smithy.example
+
+        use aws.cloudformation#excludeProperty
+        use aws.cloudformation#resource
+
+        @resource
+        resource Foo {
+            identifiers: {
+                fooId: String,
+            },
+            get: GetFoo,
+        }
+
+        @http(method: "GET", uri: "/foos/{fooId}", code: 200)
+        operation GetFoo {
+            input: GetFooRequest,
+            output: GetFooResponse,
+        }
+
+        structure GetFooRequest {
+            @httpLabel
+            @required
+            fooId: String,
+        }
+
+        structure GetFooResponse {
```

Comment:
Is this only used on read operations? Can it be used anywhere in the closure of the output? What about input?